src/markdown.c

```diff
@@ -1744,7 +1742,7 @@ parse_listitem(struct buf *ob, struct sd_markdown *rndr, uint8_t *data, size_t s
 			if (in_empty)
 				has_inside_empty = 1;
 
-			if (pre == orgpre) /* the following item must have */
+			if (i <= orgpre) /* the following item must have */
```

Comment:
Why are you worried about that? You have a `break` in the next line. So, any other text after this doesn't need to be talked about.